FuelLabs/fuel-indexer
# enhancement: forc index status: report indexer status and errors

### Description

Closes #1316.

This PR adds information about active indexers to `forc index status`.

### Testing steps

Start Fuel Indexer Service:
```
cargo run -p fuel-indexer -- run --fuel-node-host beta-4.fuel.network --fuel-node-port 80 --replace-indexer
```

Modify `fuel-explorer` to trigger a panic:

```
    fn index_block(block_data: BlockData) {
        let _x: Option<usize> = None.unwrap();
        ...
    }
```

Deploy:
```
forc-index deploy --path examples/fuel-explorer/fuel-explorer
```

Deploy another indexer:
```
forc-index deploy --path examples/hello-world/hello-world/
```

Check the status

```
cargo run -p forc-index -- status
```

![Screenshot 2023-10-17 at 5 09 32 PM](https://github.com/FuelLabs/fuel-indexer/assets/1270689/9d601ed9-b740-4ce0-b52f-741856f43ff2)


`fuel-explorer`'s status should be an `error` while the other indexer's status should be `running`, and the status message should show an increasing block count.

### Changelog

* Add additional information to `forc index status`
* Add `IndexerStatus` and `IndexerStatusKind` types
* Create `index_status` table to hold the new information
* Add `humantime` formatting of `created at`, e.g. `(23 min 30s ago)`
